src/components/measureGroups/MeasureGroups.tsx

```diff
@@ -640,114 +640,141 @@ const MeasureGroups = () => {
+                            formik.setFieldValue(
+                              `stratifications[${i}].association`,
+                              e.target.value
+                            );
+                          }}
+                          InputLabelProps={{ shrink: false }}
+                          SelectProps={{
+                            native: true,
+                          }}
+                          name="type"
+                        >
+                          {formik.values.scoring != "Select" &&
+                            Object.values(
+                              AssociationSelect[formik.values.scoring]
+                            ).map((opt, i) => (
+                              <option
+                                key={`${opt}-${i}`}
+                                value={`${opt}`}
+                                data-testid="association-select-option"
+                              >
+                                {opt}
+                              </option>
+                            ))}
+                        </TextField>
+                      </Col>
+                      <Col>
+                        <FieldLabel htmlFor="stratification-description">
+                          Stratification {i + 1} Description
+                        </FieldLabel>
+                        <FieldSeparator>
+                          {canEdit && (
+                            <textarea
+                              value={
+                                formik.values.stratifications[i].description
+                              }
+                              //type="text"
+                              name="stratification-description"
+                              id="stratification-description"
+                              autoComplete="stratification-description"
+                              placeholder="Enter Description"
+                              data-testid="stratificationDescriptionText"
+                              maxLength={5000}
+                              {...formik.getFieldProps(
+                                `stratifications[${i}].description`
+                              )}
+                            />
+                          )}
+                        </FieldSeparator>
+                      </Col>
+                    </Row>
+                    <Divider />
+                  </div>
                 ))}
+                <Row>
+                  <Button
+                    buttonTitle="Add Stratification"
+                    data-testid="add-strat-button"
+                    onClick={(e) => {
+                      e.preventDefault();
+                      formik.values.stratifications = [
+                        ...formik.values.stratifications,
+                        EmptyStrat,
+                      ];
+
+                      //idk how to force a component update in a less dumb way
+                      if (successMessage !== "") {
```

Comment:
Will leave up to you but I feel like this is a good usecase of Formik FieldArray. it gives you bunch of helpers that you can use to add/remove form fields(avoid manual handling) :
https://codesandbox.io/s/formik-fieldarray-forked-lyygcg?file=/src/index.js:1660-1668

https://formik.org/docs/api/fieldarray

Reply:
I intentionally designed it for 4511 with removing stratifications. I'll do research on this and if it's better I'll change it when both are integrated.